pandas/core/indexes/category.py

```diff
@@ -280,6 +281,30 @@ def _is_dtype_compat(self, other) -> Categorical:
 
         return other
 
+    @doc(Index.astype)
+    def astype(self, dtype, copy: bool = True) -> Index:
+        from pandas.core.api import NumericIndex
```

Comment:
can annotate dtype as Dtype

Reply:
Ok.

Reply:
should we change/deprecate Index.astype(np.int32), also Index(foo, dtype=np.int32)?

Reply:
Hmm, Index.astype(np.int32) returning an Int64Index is probably used internally, so a deprecation would require if/else clauses at various locations in the code base. Probably possible, but could we get this PR in first? This PR isn't public-facing, so donøt see a problem just merging this first.

Reply:
doing that as a follow-up sounds good